src/backend/distributed/shardsplit/shardsplit_decoder.c

```diff
@@ -73,11 +141,15 @@ static void
 split_change_cb(LogicalDecodingContext *ctx, ReorderBufferTXN *txn,
 				Relation relation, ReorderBufferChange *change)
 {
+	//elog(LOG,"Citus split_change_cb called");
```

Comment:
Sharing one comment from an internal chat for both visibility (and not to forget):


_Semantically speaking, it should probably be skipped on the target, because when that change is published, the primary data is still on the source shard (e.g., that is the one in the visible metadata).
But I guess it is also fine to skip changes on the source as well. I think in your patch we skip publishing on the source (e.g., output plugin). I don't think there is any correctness issues, but just thinking loudly if there are any edge cases this is wrong (or confusing)_

Reply:
>  I don't think there is any correctness issues,

Hmm, actually, what happens if shard split fails/cancelled at a later stage? We would have skipped publishing these changes?

Reply:
Ah, never mind, `filter_by_origin_cb ` is called on the target. (I was confused at first given `change_cb ` is called on the source.)